source/common/router/upstream_request.cc

```diff
@@ -41,18 +41,190 @@
 namespace Envoy {
 namespace Router {
 
+// This is the last stop in the filter chain: take the headers and ship them to the codec.
```

Comment:
This is the start of a big block of newly added code. Is it copied from somewhere else, or is this all new?

Reply:
this is all new

Reply:
I don't think there's much more code sharing that can be done.
Last filter in the downstream chain is the router, and that's pretty different from the codec.